File: code/dynamic_programming/factorial/factorial.js
// Part of Cosmos by OpenGenus Foundation 

function factorial(num) {
  if (num == 0) return 1;
  
  return num * factorial (num - 1);
}

let testVal = 5;
let fact = factorial(testVal);

console.log(`${testVal} factorial is ${fact}.`);

File: cosmos-master/code/mathematical-algorithms/factorial/Factorial.java
import java.util.Scanner;
// Part of Cosmos by OpenGenus Foundation
public class Factorial {
    public static void main(String args[]) {
        Scanner stdin = new Scanner(System.in);
        Factorial obj = new Factorial();
        System.out.println("Enter a number");
        int n = stdin.nextInt();
        System.out.println("The factorial is " + obj.fact(n));
    }
    
    long fact(int n) {
        if(n == 0) {
            return 1;
        } else {
            return ((long)n * fact(n-1));
        }
    }
}
